xarray/core/accessors.py

```diff
@@ -256,3 +260,888 @@ def round(self, freq):
             Array-like of datetime fields accessed for each element in values
         '''
         return self._tslib_round_accessor("round", freq)
+
+
+def _is_str_like(x):
+    return isinstance(repl, str) or isinstance(repl, bytes)
+
+
+class StringAccessor:
+    """Vectorized string functions for string-like arrays.
+
+    Similar to pandas, fields can be accessed through the `.str` attribute
+    for applicable DataArrays.
+
+        >>> da = xr.DataArray(['some', 'text', 'in', 'an', 'array'])
+        >>> ds.str.len()
+        <xarray.DataArray (dim_0: 5)>
+        array([4, 4, 2, 2, 5])
+        Dimensions without coordinates: dim_0
+
+    """
+
+    def __init__(self, xarray_obj):
+        self._obj = xarray_obj
+
+    def _apply(self, f):
+        # TODO handling of na values ?
+        return apply_ufunc(np.vectorize(f), self._obj)
```

Comment:
Can you add `dask='parallelized` to add dask support? I'm pretty sure that should "just work" with these functions.